From 6ea5ca336d9ea548c0a0fdb78d7db4f8f91b1670 Mon Sep 17 00:00:00 2001
From: Jiri Popelka <jpopelka@redhat.com>
Date: Fri, 15 Aug 2014 15:15:32 +0200
Subject: [PATCH] lib/dhcpsrv Makefile.am: move dhcp_data_dir @localstatedir@
 -> @sharedstatedir@
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

On Fedora @localstatedir@ resolves to /var/ and @sharedstatedir@ to /var/lib/.

FHS says [1] that:
"Applications must generally not add directories to the top level of /var.
Such directories should only be added if they have some system-wide implication,
and in consultation with the FHS mailing list."

From GNU Coding Standards [2]:
‘sharedstatedir’
    The directory for installing architecture-independent data files
which the programs modify while they run. This should normally be
/usr/local/com, but write it as $(prefix)/com. (If you are using
Autoconf, write it as ‘@sharedstatedir@’.)

‘localstatedir’
    The directory for installing data files which the programs modify
while they run, and that pertain to one specific machine. Users should
never need to modify files in this directory to configure the package’s
operation; put such configuration information in separate files that go
in $(datadir) or $(sysconfdir). $(localstatedir) should normally be
/usr/local/var, but write it as $(prefix)/var. (If you are using
Autoconf, write it as ‘@localstatedir@’.)

[1] http://www.pathname.com/fhs/pub/fhs-2.3.html#THEVARHIERARCHY
[2] https://www.gnu.org/prep/standards/html_node/Directory-Variables.html
---
 src/lib/dhcpsrv/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/lib/dhcpsrv/Makefile.am b/src/lib/dhcpsrv/Makefile.am
index b10d269..0adc408 100644
--- a/src/lib/dhcpsrv/Makefile.am
+++ b/src/lib/dhcpsrv/Makefile.am
@@ -1,6 +1,6 @@
 SUBDIRS = . testutils tests
 
-dhcp_data_dir = @localstatedir@/@PACKAGE@
+dhcp_data_dir = @sharedstatedir@/@PACKAGE@
 
 AM_CPPFLAGS = -I$(top_builddir)/src/lib -I$(top_srcdir)/src/lib -DDHCP_DATA_DIR="\"$(dhcp_data_dir)\""
 AM_CPPFLAGS += -DTOP_BUILDDIR="\"$(top_builddir)\""
-- 
1.9.3

